main.ts

```diff
@@ -84,6 +93,136 @@ if ($searchInput) {
+    '.characterVideogames',
+  ) as HTMLElement;
+  const $characterInfo = document.querySelector(
+    '.characterInfo',
+  ) as HTMLElement;
+
+  $characterPic.src = charData.imageUrl;
+  $characterPic.alt = charData.name;
+  $characterFilm.textContent = charData.films.join(' , ');
+  $characterVideogames.textContent = charData.videoGames.join(' ');
+  $characterInfo.innerHTML = `<a href="${charData.sourceUrl}">${charData.sourceUrl}</a>`;
+
+  // main page swapping
+  const $mainPage = document.querySelector(
+    '[data-view="main-page"]',
+  ) as HTMLElement;
+  const $characterDetail = document.querySelector(
+    '[data-view="character-page"]',
+  ) as HTMLElement;
+
+  $mainPage.classList.add('hidden');
+  $characterDetail.classList.remove('hidden');
+}
+
+// Disney header navbar, one way ticket back to main page
+
+const $navbar = document.querySelector('.navbar') as HTMLElement;
+
+if ($navbar) {
+  $navbar.addEventListener('click', () => {
+    // this is 2 data-view
+    const $mainPage = document.querySelector('[data-view="main-page"]');
+    const $characterPage = document.querySelector(
+      '[data-view="character-page"',
+    );
+
+    // click on Walt Disney = go back to main page
+    $characterPage?.classList.add('hidden');
+    $mainPage?.classList.remove('hidden');
+  });
+}
+
 document.addEventListener('DOMContentLoaded', () => {
   disneyData();
 });
+
+// This is Where the Fun Begin, Here is all the local storage code for dislike and favorite start
+
+// first part will be code in Character-Page
+
+// set how data going into  and out of local storage
+function saveToLocalStorage(key: string, data: Data[]): void {
+  localStorage.setItem(key, JSON.stringify(data));
+}
+
+function getFromLocalStorage(key: string): Data[] {
+  const data = localStorage.getItem(key);
+  return data ? JSON.parse(data) : [];
+}
+
+// function to display character's data
+
+function getCharData(): Data {
+  const $characterPic = document.querySelector(
+    '#character-image',
+  ) as HTMLImageElement;
+  const $characterFilm = document.querySelector(
+    '.characterFilm',
+  ) as HTMLElement;
+  const $characterVideogames = document.querySelector(
+    '.characterVideogames',
+  ) as HTMLElement;
+  const $characterInfo = document.querySelector(
+    '.characterInfo',
+  ) as HTMLElement;
+
+  return {
+    _id: Date.now(), // Temporary unique ID for demonstration
+    name: $characterPic.alt,
+    imageUrl: $characterPic.src,
+    createdAt: new Date().toISOString(),
+    updatedAt: new Date().toISOString(),
+    films: $characterFilm.textContent
+      ? $characterFilm.textContent.split(', ')
+      : [],
+    shortFilms: [],
+    tvShows: [],
+    videoGames: $characterVideogames.textContent
+      ? $characterVideogames.textContent.split(' ')
+      : [],
+    allies: [],
+    enemies: [],
+    parkAttractions: [],
+    sourceUrl: $characterInfo.querySelector('a')?.href || '',
+    url: '',
+  };
+}
```

Comment:
What is this supposed to accomplish? If you ever need to get characters data, you should be getting it from the API. So you should be making another fetch call for the specific character you want by their id